homeassistant/components/ecovacs/sensor.py

```diff
@@ -0,0 +1,269 @@
+"""Ecovacs sensor module."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from math import floor
+from typing import Any, Generic
+
+from deebot_client.capabilities import CapabilityEvent, CapabilityLifeSpan
+from deebot_client.device import Device
+from deebot_client.events import (
+    BatteryEvent,
+    ErrorEvent,
+    Event,
+    LifeSpan,
+    LifeSpanEvent,
+    NetworkInfoEvent,
+    StatsEvent,
+    TotalStatsEvent,
+)
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    AREA_SQUARE_METERS,
+    ATTR_BATTERY_LEVEL,
+    CONF_DESCRIPTION,
+    PERCENTAGE,
+    EntityCategory,
+    UnitOfTime,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .const import DOMAIN
+from .controller import EcovacsController
+from .entity import (
+    EcovacsCapabilityEntityDescription,
+    EcovacsDescriptionEntity,
+    EcovacsEntity,
+    EventT,
+)
+from .util import get_supported_entitites
+
+
+@dataclass(kw_only=True, frozen=True)
+class EcovacsSensorEntityDescription(
+    EcovacsCapabilityEntityDescription,
+    SensorEntityDescription,
+    Generic[EventT],
+):
+    """Ecovacs sensor entity description."""
+
+    value_fn: Callable[[EventT], StateType]
+
+
+ENTITY_DESCRIPTIONS: tuple[EcovacsSensorEntityDescription, ...] = (
+    # Stats
+    EcovacsSensorEntityDescription[StatsEvent](
+        key="stats_area",
+        capability_fn=lambda caps: caps.stats.clean,
+        value_fn=lambda e: e.area,
+        translation_key="stats_area",
+        native_unit_of_measurement=AREA_SQUARE_METERS,
+    ),
+    EcovacsSensorEntityDescription[StatsEvent](
+        key="stats_time",
+        capability_fn=lambda caps: caps.stats.clean,
+        value_fn=lambda e: round(e.time / 60) if e.time is not None else None,
+        translation_key="stats_time",
+        native_unit_of_measurement=UnitOfTime.MINUTES,
```

Comment:
```suggestion
        value_fn=lambda e: e.time,
        translation_key="stats_time",
        native_unit_of_measurement=UnitOfTime.SECONDS,
        suggested_unit_of_measurement=UnitOfTime.MINUTES,
```

Reply:
Each minute we get an update from the bot so not sure if we should use seconds as maybe the user gets confused as the values is not counting upwords...